src/framework/linalg/almost_equal.hpp

```diff
@@ -39,6 +39,26 @@ bool almost_equal(T f1, T f2,
          max_relative_diff * std::max(std::abs(f1), std::abs(f2));
 }
 
+// Complex numbers are probably not considered scalar, so we need another
```

Comment:
Complex numbers are considered scalar, scalar just means not-matrix/vector valued